### DISCUSSION TITLE
Does code-server supports RISC-V architecture

### DISCUSSION THREAD

**Discussion by sakthivel-axim** [coder/code-server#6266] [Q&A]
Do we have supported package for RISC-V architecture is there any compilation steps to generate package

**Comment by code-asher** [comment]
There is nothing official available, but if there is a Node build 
for RISC-V maybe running `npm install code-server` will work. 
Might need `export npm_config_build_from_source=true` to build 
from source.

**Reply by sakthivel-axim** [reply]
Ok, Thanks @code-asher

**Comment by karthick-govindaraj** [comment]
Hi @code-asher ,

I have found unofficial Node.js builds for RISC-V on their website at https://unofficial-builds.nodejs.org/download/release/v18.19.0/. Additionally, for Code Server, which is built on Electron.js, we found Electron builds for RISC-V at https://github.com/riscv-forks/electron-riscv-releases/releases.

Do you have any plans to develop support for Code Server on RISC-V? If so, when can we expect this support to be available in the future any specific timeline?

**Reply by code-asher** [reply]
Just FYI, code-server depends only on Node, not Electron.

We have no plans on adding releases for RISC-V so you would need to build from source.

I think we could revisit if Node adds RISC-V as an officially supported platform though.

**Comment by benz0li** [comment]
@code-asher I have created an issue addressing this topic: https://github.com/coder/code-server/issues/6951

**Reply by code-asher** [reply]
Thank you!
